### DISCUSSION TITLE
[Webpack 4] Confused about output.chunkFilename

### DISCUSSION THREAD

**Discussion by JinHe-PerkinElmer** [webpack/webpack#16489] [Q&A]
I am very confused about output.chunkFilename now, here is part of my webpack config js:

output: {

    filename: isEnvProduction
      ? 'static/normal/js/[name].[contenthash:8].js'
      : isEnvDevelopment && 'static/js/bundle.js',

    // There are also additional JS chunk files if you use code splitting.
    chunkFilename: isEnvProduction
      ? 'static/normal/js/lazy-[name].[contenthash:8].chunk.js'
      : isEnvDevelopment && 'static/js/lazy-[name].chunk.js',

  },
I'm using React.lazy and React.Suspense for better app performance, I try to distinguish the file name of the webpack asynchronous chunks. Very simple, all asynchronous chunks are prefixed with lazy.

Confusingly, all chunks are prefixed with lazy after webpack production build, but when I run development build, it works as I wish. So Has anyone encountered this before, any ideas? @alexander-akait

**Comment by alexander-akait** [comment]
Not sure, maybe splitting in merge/split some chunks, so name is overriden, I need more infromation - configuration

**Reply by JinHe-PerkinElmer** [reply]
Oh, I think I found the cause of the problem, I am using optimization.splitChunks like this: 
`splitChunks: {
        chunks: 'all',
        name: false,
      },`
So should I use `optimization.splitChunks.chunks='async'`? Do you have any suggestions, thank you！

**Reply by alexander-akait** [reply]
`splitChunks` have the `filename` option, so you can give a new there, webpack v5 have better logic here, unfortunately we can't backport it

**Reply by JinHe-PerkinElmer** [reply]
I checked the official documentation of V4, there are two related configurations.
 First is **`splitChunks.name`**: https://v4.webpack.js.org/plugins/split-chunks-plugin/#splitchunksname
Second is **`splitChunks.cacheGroups.{cacheGroup}.filename`**: https://v4.webpack.js.org/plugins/split-chunks-plugin/#splitchunkscachegroupscachegroupfilename.
_**Do you mean the second one?**_ 
But now my splitChunks configuration is `splitChunks: { chunks: 'all', name: false, }`,  I think it's a bit difficult to distinguish dynamically imported filenames, but I'll give it a try, thanks very much!
